### PR TITLE
Proof of concept

### PR DESCRIPTION
**Pull request by HectorPeeters**
# Description

Merge into master after sprint week 5.

# Deletes Source Branch?

Yes:    Work is done, the branch can be safely deleted.

# How Many Approvals?

* Approval of all (Controversial changes that change the entire project)

### CODE REVIEW

**Review by FBroy — APPROVED** [review]

**Review by yuanzexiong — CHANGES_REQUESTED** [review]
Let's wait until the other small feature pull requests come in and then we can merge this one to master and check it in to Fonto's repo.

**Review by yuanzexiong — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -66,6 +67,9 @@ function parseXPath(xpathString: string) {
 	}
 
 	const ast = parseExpression(xpathString, { allowXQuery: false });
+
+	annotateAst(ast);
```
Comment (maanpalle) [inline src/index.ts:71]: index should not be changed

Comment (HectorPeeters) [reply src/index.ts:71]: Were just adding annotation here so that should be fine

Comment (FBroy) [reply src/index.ts:71]: The imports should not be changed, our content can be changed.

**Review by maanpalle — APPROVED** [review]

File: .github/workflows/bundlesize.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@ jobs:
               run: echo "CI_COMMIT_SHA=${{ github.event.pull_request.head.sha}}" >> $GITHUB_ENV
 
             - name: Run BundleMon
+              if: secrets.BUNDLEMON_PROJECT_APIKEY
```
Comment (MTimmerije) [inline .github/workflows/bundlesize.yml:26]: At some point, do we want a different solution for this?

File: src/parsing/parseExpression.ts [hunk 3]
```diff
@@ -37,6 +37,7 @@ export default function parseExpression(
 				storeParseResultInCache(xPathString, language, ast);
 			}
 		}
+
```
Comment (MTimmerije) [inline src/parsing/parseExpression.ts:40]: Nice

**Review by MTimmerije — APPROVED** [review]
I don't see any real problems, minus that the package size solution seems like a band aid (but we can't do anything about that), so it gets my stamp of approval.

**Pull request merged**